src/main/java/com/dreammaster/baubles/OvenGlove.java

```diff
@@ -210,6 +211,12 @@ public void onDamage(LivingAttackEvent event) {
             }
             EntityPlayer player = (EntityPlayer) entity;
 
+            // If the game is in peaceful, exit early. Why do this? See issue #16879
```

Comment:
Move to after the `DamageSourceHotItem` check

Reply:
Damn, done. Not sure how i missed that my bad.